torch_audiomentations/augmentations/background_noise.py

```diff
@@ -4,113 +4,143 @@
-        device=torch.device("cpu"),
         mode: str = "per_example",
         p: float = 0.5,
-        p_mode: typing.Optional[str] = None,
-        sample_rate: typing.Optional[int] = None,
+        p_mode: str = None,
+        sample_rate: int = None,
     ):
-        # TODO: infer device from the given samples instead
-        super(ApplyBackgroundNoise, self).__init__(mode, p, p_mode, sample_rate)
-        self.bg_path = find_audio_files(bg_path)
+        """
+
+        :param background_paths: list of paths to background audio files. 
+        :param min_snr_in_db: minimum SNR in dB. 
+        :param max_snr_in_db: maximium SNR in dB.
+        :param mode:
+        :param p:
+        :param p_mode:
+        """
+
+        super().__init__(mode, p, p_mode, sample_rate)
+
+        if isinstance(background_paths, (list, tuple, set)):
+            # TODO: check that one can read audio files
+            self.background_paths = list(background_paths)
+        else:
+            self.background_paths = find_audio_files(background_paths)
 
-        if len(self.bg_path) == 0:
+        if sample_rate is not None:
+            self.audio = Audio(sample_rate=sample_rate, mono=True)
+
+        if len(self.background_paths) == 0:
             raise EmptyPathException("There are no supported audio files found.")
 
         self.min_snr_in_db = min_snr_in_db
         self.max_snr_in_db = max_snr_in_db
+        if self.min_snr_in_db > self.max_snr_in_db:
+            raise ValueError("min_snr_in_db must not be greater than max_snr_in_db")
         self.snr_distribution = torch.distributions.Uniform(
             low=min_snr_in_db, high=max_snr_in_db, validate_args=True
         )
-        self.device = device
 
-    def randomize_parameters(
-        self, selected_samples, sample_rate: typing.Optional[int] = None
-    ):
-        selected_batch_size = selected_samples.size(0)
-        bg_file_paths = random.choices(self.bg_path, k=selected_batch_size)
-        bg_audios = []
-        for index, bg_file_path in enumerate(bg_file_paths):
-            bg_audio_info = soundfile.info(bg_file_path, verbose=True)
-            bg_audio_num_samples = math.ceil(
-                sample_rate * bg_audio_info.frames / bg_audio_info.samplerate
-            )
-            samples_num_samples = selected_samples[index].size(0)
-
-            # ensure that background noise has the same length as the sample
-            if bg_audio_num_samples < samples_num_samples:
-                bg_start_index = 0
-                bg_stop_index = bg_audio_info.frames
-                loaded_bg_audio_num_samples = bg_audio_num_samples
-
-                current_bg_audio = load_audio(
-                    bg_file_path,
-                    sample_rate=sample_rate,
-                    start=bg_start_index,
-                    stop=bg_stop_index,
-                )
-                bg_audio = [current_bg_audio]
+    def rms(self, selected_samples: torch.Tensor) -> torch.Tensor:
```

Comment:
Maybe update (set keepdims=False) `calculate_rms` in `torch_audiomentations.utils.dsp`, and use that instead? I can see how calculating the RMS is useful in other transforms as well

Reply:
OK. I wasn't sure whether RMS would be used elsewhere. Will do that.

Reply:
Done.